src/einsteinpy/hypersurface/schwarzschildembedding.py

```diff
@@ -0,0 +1,194 @@
+import matplotlib.pyplot as plt
+import numpy as np
+from astropy import units as u
+from mpl_toolkits import mplot3d
+
+
+class SchwarzschildEmbedding:
+    """
+    Class for Utility functions for Schwarzschild Embedding surface to
+    implement gravitational lensing
+
+    Attributes
+    ----------
+    input_units : list
+                  list of input units of M
+    units_list : list
+                 customized units to handle values of M and render plots
+                 within grid range
+    r_init : ~astropy.units.m
+
+    """
+
+    def __init__(self, M):
+        """
+      Constructor
+      Initialize mass and embedding initial radial coordinate in appropiate units
+      in order to render the plots of the surface in finite grid. The initial r
+      is taken to be just greater than schwarzschild radius but it is important
+      to note that the embedding breaks at r < 9m/4.
+
+      Parameters
+      ----------
+      M : ~astropy.units.kg
+          Mass of the body
+
+      """
+        self.input_units = [M.unit]
+        self.units_list = [u.kg * 10e22, u.m / M.to(u.kg * 10e22).value]
+        M = M.to(self.units_list[0])
+        self.M = M
+        self.r_init = (((3 * self.M.value + 0.0001) / self.M.value) * u.m).to(
+            self.units_list[1]
+        )
+
+    def gradient(self, r):
+        """
+      Calculate gradient of Z coordinate w.r.t r to update the value of r and
+      thereby get value of spherical radial coordinate R.
+
+      Parameters
+      ----------
+      r : float
+          schwarzschild coordinate at which gradient is supposed to be obtained
+
+      Returns
+      -------
+      float : gradient of Z w.r.t r at the point r (passed as argument)
+
+      """
+        R = r / np.sqrt(1 - (2 * self.M.value / r))
+        num_one = 1 - (3 * self.M.value / r)
+        num_two = np.sqrt(
+            ((4 * self.M.value * r - 9 * self.M.value * self.M.value) * R)
+            / (r - 3 * self.M.value) ** 2
+        )
+        deno = np.sqrt(1 - (2 * self.M.value / r)) ** 3
+
+        return num_one * num_two / deno
+
+    def radial_coord(self, r):
+        """
+      Returns spherical radial coordinate (of the embedding) from given schwarzschild
+      coordinate.
+
+      Parameters
+      ----------
+      r : float
+
+      Returns
+      -------
+      float : spherical radial coordinate of the 3d embedding
```

Comment:
```
float
    spherical radial coordinate of 3d embedding
```

Reply:
One question, is this function calculating `theta`, given a `r`  ?

Reply:
This is calculating R (embedding radial coordinate as given in above docstring) for a given r (schwarzschild radial coordinate).